gateway-release/home/conf/topologies/metadata.xml

```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="utf-8"?>
```

Comment:
Is this topology just an example? Does it need to be exposed by default? If its an example `gateway-release/home/templates/` would be a better place.

Reply:
No, it's not. Without this topology, clients (CM, Ambari, etc...) can not query metadata (general proxy info and topology information).
Please note this can not be added into Admin API because there is no need to authorize the incoming requests.

Reply:
I see that the metadata.xml topology is aligned with other existing topologies like admin.xml for instance. I think that we need to consider moving these to descriptors that point to the default-providers.json shared provider config. This way the authentication can be changed in one place. Admin will require an authorization provider but since it is for the admin service and not others that should be okay. Let's not block this PR and follow up with another to move to descriptors. Just wanted to mention it here.

Reply:
https://issues.apache.org/jira/browse/KNOX-2251

Reply:
https://issues.apache.org/jira/browse/KNOX-2251